Dockerfile

```diff
@@ -30,6 +21,15 @@ RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
 # See: rust-lang.org/tools/install (Configuring the PATH environment variable).
 ENV PATH="/root/.cargo/bin:${PATH}"
 
+# Copy code into the container.
+RUN mkdir /code
+COPY ./ /code
```

Comment:
I'd suggest moving all this even further down, after line 31 (install pip and setuptools), as none of the lines before depend on the code.